x/fundraising/keeper/hooks_test.go

```diff
@@ -0,0 +1,188 @@
+	BeforeBidPlacedValid                bool
+	BeforeBidModifiedValid              bool
+	BeforeAllowedBiddersAddedValid      bool
+	BeforeAllowedBidderUpdatedValid     bool
+}
+
+func (h *MockFundraisingHooksReceiver) BeforeFixedPriceAuctionCreated(
+	ctx sdk.Context,
+	auctioneer string,
+	startPrice sdk.Dec,
+	sellingCoin sdk.Coin,
+	payingCoinDenom string,
+	vestingSchedules []types.VestingSchedule,
+	startTime time.Time,
+	endTime time.Time,
+) {
+	h.BeforeFixedPriceAuctionCreatedValid = true
+}
+
+func (h *MockFundraisingHooksReceiver) BeforeBatchAuctionCreated(
+	ctx sdk.Context,
+	auctioneer string,
+	startPrice sdk.Dec,
+	minBidPrice sdk.Dec,
+	sellingCoin sdk.Coin,
+	payingCoinDenom string,
+	vestingSchedules []types.VestingSchedule,
+	maxExtendedRound uint32,
+	extendedRoundRate sdk.Dec,
+	startTime time.Time,
+	endTime time.Time,
+) {
+	h.BeforeBatchAuctionCreatedValid = true
+}
+
+func (h *MockFundraisingHooksReceiver) BeforeAuctionCanceled(
+	ctx sdk.Context,
+	auctionId uint64,
+	auctioneer string,
+) {
+	h.BeforeAuctionCanceledValid = true
+}
+
+func (h *MockFundraisingHooksReceiver) BeforeBidPlaced(
+	ctx sdk.Context,
+	auctionId uint64,
+	bidder string,
+	bidType types.BidType,
+	price sdk.Dec,
+	coin sdk.Coin,
+) {
+	h.BeforeBidPlacedValid = true
+}
+
+func (h *MockFundraisingHooksReceiver) BeforeBidModified(
+	ctx sdk.Context,
+	auctionId uint64,
+	bidder string,
+	bidType types.BidType,
+	price sdk.Dec,
+	coin sdk.Coin,
+) {
+	h.BeforeBidModifiedValid = true
+}
+
+func (h *MockFundraisingHooksReceiver) BeforeAllowedBiddersAdded(
+	ctx sdk.Context,
+	auctionId uint64,
+	allowedBidders []types.AllowedBidder,
+) {
+	h.BeforeAllowedBiddersAddedValid = true
+}
+
+func (h *MockFundraisingHooksReceiver) BeforeAllowedBidderUpdated(
+	ctx sdk.Context,
+	auctionId uint64,
+	bidder sdk.AccAddress,
+	maxBidAmount sdk.Int,
+) {
+	h.BeforeAllowedBidderUpdatedValid = true
+}
+
+func (s *KeeperTestSuite) TestHooks() {
+	fundraisingHooksReceiver := MockFundraisingHooksReceiver{}
+
+	// Set hooks
+	s.keeper.SetHooks(types.NewMultiFundraisingHooks(&fundraisingHooksReceiver))
+
+	s.Require().False(fundraisingHooksReceiver.BeforeFixedPriceAuctionCreatedValid)
+	s.Require().False(fundraisingHooksReceiver.BeforeBatchAuctionCreatedValid)
+	s.Require().False(fundraisingHooksReceiver.BeforeAuctionCanceledValid)
+	s.Require().False(fundraisingHooksReceiver.BeforeBidPlacedValid)
+	s.Require().False(fundraisingHooksReceiver.BeforeBidModifiedValid)
+	s.Require().False(fundraisingHooksReceiver.BeforeAllowedBiddersAddedValid)
+	s.Require().False(fundraisingHooksReceiver.BeforeAllowedBidderUpdatedValid)
+
+	// Create a fixed price auction
+	s.createFixedPriceAuction(
+		s.addr(0),
+		parseDec("2.0"),
+		parseCoin("1_000_000_000_000denom1"),
+		"denom2",
+		[]types.VestingSchedule{},
+		time.Now().AddDate(0, 0, -1),
+		time.Now().AddDate(0, 0, -1).AddDate(0, 2, 0),
+		true,
+	)
+	s.Require().True(fundraisingHooksReceiver.BeforeFixedPriceAuctionCreatedValid)
```

Comment:
Here's some just suggestion. How about adding the other checking parts here to show that the other Booleans are not changed?
        s.Require().False(fundraisingHooksReceiver.BeforeBatchAuctionCreatedValid)
	s.Require().False(fundraisingHooksReceiver.BeforeAuctionCanceledValid)
	s.Require().False(fundraisingHooksReceiver.BeforeBidPlacedValid)
	s.Require().False(fundraisingHooksReceiver.BeforeBidModifiedValid)
	s.Require().False(fundraisingHooksReceiver.BeforeAllowedBiddersAddedValid)
	s.Require().False(fundraisingHooksReceiver.BeforeAllowedBidderUpdatedValid)
This can be also added below. But, I'm not sure how much this helps in testing.

Reply:
Well, then we need to put all those `False` lines for every validation. I think it is redundant since they are validated to be false in the first place. To provide how this test case is written, testing methodology is referenced from Cosmos SDK [x/gov](https://github.com/cosmos/cosmos-sdk/blob/master/x/gov/keeper/hooks_test.go).